src/Sparrow/ByteString.cs

```diff
@@ -590,7 +590,31 @@ public void LowMemoryOver()
 
         public class SegmentStack : StackHeader<UnmanagedGlobalSegment>
         {
+            ~SegmentStack()
+            {
+                if (Environment.HasShutdownStarted)
+                    return; // no need
 
+                try
+                {
+                    var current = Interlocked.Exchange(ref Head, HeaderDisposed);
+                    while (current != null)
+                    {
+                        var segment = current.Value;
+                        current = current.Next;
+                        if (segment == null)
+                            continue;
+                        if (!segment.InUse.Raise())
+                            continue;
+                        segment.Dispose();
+                    }
+                }
+                catch (Exception)
```

Comment:
This is wrong, we should _never_ throw from finalizer.

Reply:
There is no difference between this finalizer and all other finalizers in that file. Other finalizer prevent throwing "ObjectDisposedException" but other exception won't be treated. And frankly, if you have memory management exception during finalization, you better loose the process then continue working in invalid memory state.